control-plane/agents/src/bin/core/tests/volume/switchover.rs

```diff
@@ -0,0 +1,110 @@
+use common_lib::types::v0::transport::{
+    CreateVolume, Filter, PublishVolume, RepublishVolume, VolumeShareProtocol,
+};
+use deployer_cluster::ClusterBuilder;
+use grpc::operations::{replica::traits::ReplicaOperations, volume::traits::VolumeOperations};
+use std::time::Duration;
+use tokio::time::sleep;
+
+const VOLUME_UUID: &str = "1e3cf927-80c2-47a8-adf0-95c486bdd7b7";
+const POOL_SIZE_BYTES: u64 = 128 * 1024 * 1024;
+
+#[tokio::test]
+async fn volume_republish_nexus_recreation() {
+    let cluster = ClusterBuilder::builder()
+        .with_rest(false)
+        .with_agents(vec!["core"])
+        .with_io_engines(2)
+        .with_tmpfs_pool(POOL_SIZE_BYTES)
+        .with_cache_period("1s")
+        // don't let the reconcile interfere with the tests
+        .with_reconcile_period(Duration::from_secs(1000), Duration::from_secs(1000))
+        .build()
+        .await
+        .unwrap();
+
+    let client = cluster.grpc_client().volume();
+
+    assert!(client
+        .create(
+            &CreateVolume {
+                uuid: VOLUME_UUID.try_into().unwrap(),
+                size: 5242880,
+                replicas: 1,
+                ..Default::default()
+            },
+            None,
+        )
+        .await
+        .is_ok());
+
+    let replicas = cluster
+        .grpc_client()
+        .replica()
+        .get(Filter::None, None)
+        .await
+        .expect("error getting replicas")
+        .into_inner();
+
+    let replica_node = replicas
+        .get(0)
+        .expect("Should have one replica")
+        .node
+        .as_str();
+
+    let volume = client
+        .publish(
+            &PublishVolume {
+                uuid: VOLUME_UUID.try_into().unwrap(),
+                share: Some(VolumeShareProtocol::Nvmf),
+                target_node: Some(replica_node.into()),
+            },
+            None,
+        )
+        .await
+        .expect("Volume publish should have succeeded.");
+
+    let older_nexus = volume
+        .state()
+        .target
+        .expect("Target should be present as publish succceded");
+
+    // Kill the node that hosts the nexus and only replica.
+    cluster
+        .composer()
+        .kill(replica_node)
+        .await
+        .expect("Node should have been killed");
+    // Wait for 5 secs for control plane refresh.
+    sleep(Duration::from_secs(5)).await;
```

Comment:
please replace these hardcoded sleeps with something that checks for whatever condition you're waiting for up to a timeout.
In this particular case, stopping rather than kill probably already achieves that as the io-engine deregisters from the control-plane on stop?

Reply:
Done.